### DISCUSSION TITLE
CREATING A MOBILE-FIRST VERSION

### DISCUSSION THREAD

**Discussion by kcelestinomaria** [near/near-wallet#1683] [Ideas]
Hello, can we create a mobile-first near wallet for users of dapps to enable seamless onboarding? I am working on a dapp and my target market is users who use their mobile phones more than their PCs, I would really love that we can work on this mobile NEAR wallet, maybe we can borrow some inspiration for how to do it from the Celo blockchain which has a mobile-first wallet to onboard users. Thank you @artob @picturepan2 @vgrichina @ilblackdragon @chadoh

**Comment by kcelestinomaria** [comment]
Also, this wallet can be downloadable on the AppStore(preferable Android). If not, maybe can we make the Mobile Wallet a PWA?

**Comment by stefanopepe** [comment]
Hi, thank you for your interest to improve NEAR Wallet. 
If you open https://wallet.near.org from a mobile phone, the interface is designed to be usable also on smaller screens.
Also, we use React for the wallet frontend, so it won't be too difficult to deploy the wallet using React Native - However, we need to implement low-level changes to the codebase, which are not planned in the short-term.

In the meantime, Apple and Android users can easily add the NEAR Wallet website as a stand-alone application, and also experience a seamless login flow to your dApps.
Let me know if the experience is acceptable, or we can already work to improve it
